### PR TITLE
Generator refactoring

### PR DESCRIPTION
**Pull request by daffl**
This is a refactoring of the current generator to accommodate the Auk release.

Still ToDo:

- [x] Model generator
- [x] User model generator
- [x] Authentication generator
- [ ] Integration tests

Addressed issues:

- Closes https://github.com/feathersjs/generator-feathers/issues/44, closes https://github.com/feathersjs/generator-feathers/issues/115 (service name inflection)
- Closes https://github.com/feathersjs/generator-feathers/issues/95 (more explicit directory structure)
- Closes https://github.com/feathersjs/generator-feathers/issues/103 (choose all database adapters)
- Closes https://github.com/feathersjs/generator-feathers/issues/120 (unhandled promise)
- Closes https://github.com/feathersjs/generator-feathers/issues/124 (duplicate `require`)
- Closes https://github.com/feathersjs/generator-feathers/issues/158 (configurable source folder)
- Closes https://github.com/feathersjs/generator-feathers/issues/94, closes https://github.com/feathersjs/feathers-sequelize/issues/63 (Sequelize associations)
- Closes https://github.com/feathersjs/feathers-cli/issues/58 (ESLint)
- Closes https://github.com/feathersjs/feathers-cli/issues/23 (RethinkDB support)
- Closes https://github.com/feathersjs/feathers-cli/issues/57 (Knex support)
- Closes https://github.com/feathersjs/feathers-cli/issues/29 (generate auth secret)
- Closes https://github.com/feathersjs/feathers-cli/issues/34 (camelCase variable names)
- Closes https://github.com/feathersjs/feathers-cli/issues/49 (no public)
- Closes https://github.com/feathersjs/feathers-cli/issues/53 (Yarn support)
- Closes https://github.com/feathersjs/feathers-cli/issues/54 (Auth 1.0)
- Closes https://github.com/feathersjs/feathers-cli/issues/56 (creating users service)
- Closes https://github.com/feathersjs/feathers-cli/issues/22 (check for updates)

### CODE REVIEW

File: generators/service/index.js [hunk 1]
```diff
@@ -1,199 +1,200 @@
...
+    const { type, kebabName } = this.props;
+    const moduleMappings = {
+      generic: `./${kebabName}.class.js`,
+      memory: 'feathers-memory',
+      nedb: 'feathers-nedb',
+      mongodb: 'feathers-mongodb',
+      mongoose: 'feathers-mongoose',
+      sequelize: 'feathers-sequelize',
+      knex: 'feathers-knex',
+      rethinkdb: 'feathers-sequelize'
```
Comment (ForsakenHarmony) [inline generators/service/index.js]: shouldn't this be feathers-rethinkdb?

Comment (corymsmith) [inline generators/service/index.js]: @ForsakenHarmony Good catch, that should definitely be `feathers-rethinkdb`

Comment (ForsakenHarmony) [inline generators/service/index.js]: was fixed already

**Comment by daffl** [comment]
This has been released as v1.0.0-pre.1

**Comment by OmgImAlexis** [comment]
@daffl I'm using the latest version of feathers-cli and it looks like npm is still being used to install the project as well as there's no `yarn.lock` being generated. 

Any ideas?

**Pull request merged**